Code/Framework/AzFramework/AzFramework/Input/Devices/Gamepad/InputDeviceGamepad.h

```diff
@@ -182,6 +182,14 @@ namespace AzFramework
         // Reflection
         static void Reflect(AZ::ReflectContext* context);
 
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        // Foward declare the internal Implementation class so it can be passed into the constructor
+        class Implementation;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! Alias for the function type used to create a custom implementation for this input device
+        using ImplementationFactory = typename Implementation*(*)(InputDeviceGamepad&);
```

Comment:
suggestion: I _think_ with the using declaration here you don't need the `(*)` anymore (this was only needed for the name when dealing with using they old `typedef` approach

Reply:
Looks like you're right, updated.